src/framework/loader.ts

```diff
@@ -1,7 +1,11 @@
+// Here and in every files we need copyright statements. BSD3 clause license like the WebGPU stuff in the group
+
 import { GroupRecorder } from './logger.js';
 import { IParamsAny, paramsEquals, paramsSupersets } from './params/index.js';
 import { allowedTestNameCharacters, ITestGroup, RunCase, ICaseID } from './test_group.js';
 
+// These data types seem super core so please add more comments
```

Comment:
good point. These are kind of a mess right now, there are way too many of them. I need to see if I can clean them up too.

Reply:
Done, added to roadmap